src/platform/packages/shared/kbn-unified-metrics-grid/src/components/metric_flyout/metrics_insights_flyout.tsx

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiText,
+  EuiFlyoutBody,
+  EuiFlyoutHeader,
+  EuiFlyoutResizable,
+  useIsWithinMinBreakpoint,
+  useEuiTheme,
+  EuiButtonEmpty,
+  EuiFlyoutFooter,
+  isDOMNode,
+  keys,
+} from '@elastic/eui';
+import React, { useCallback } from 'react';
+import useLocalStorage from 'react-use/lib/useLocalStorage';
+import type { MetricField } from '../../types';
+import { MetricFlyoutContent } from './metric_flyout_content';
+import { useFlyoutA11y } from '../../hooks/use_flyout_a11y';
+
+interface MetricInsightsFlyoutProps {
+  metric: MetricField;
+  esqlQuery: string;
+  isOpen: boolean;
+  onClose: () => void;
+}
+
+export const MetricInsightsFlyout = ({
+  metric,
+  esqlQuery,
+  isOpen,
+  onClose,
+}: MetricInsightsFlyoutProps) => {
+  const { euiTheme } = useEuiTheme();
+  const DEFAULT_WIDTH = euiTheme.base * 34;
+  const defaultWidth = DEFAULT_WIDTH; // Give enough room to search bar to not wrap
```

Comment:
Why do we need these 2 variables? Could we do this instead?

```suggestion
  const defaultWidth = euiTheme.base * 34
```